### PR TITLE
[#111] Support for EncounterRole.

### PR DESCRIPTION
**Pull request by brandones**
Fixes https://github.com/mekomsolutions/openmrs-module-initializer/issues/111

### CODE REVIEW

**Comment by brandones** [comment]
@mks-d

File: api/src/main/java/org/openmrs/module/initializer/api/er/EncounterRolesCsvParser.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	public Domain getDomain() {
+		return Domain.ENCOUNTER_ROLES;
+	}
+	
+	@Override
+	public EncounterRole bootstrap(CsvLine line) throws IllegalArgumentException {
+		
+		String uuid = line.getUuid();
+		
+		EncounterRole type = encounterService.getEncounterRoleByUuid(uuid);
```
Comment (mseaton) [inline api/src/main/java/org/openmrs/module/initializer/api/er/EncounterRolesCsvParser.java:36]: I'd probably call this variable "role" or "encounterRole"

File: api/src/test/java/org/openmrs/module/initializer/api/EncounterRolesLoaderIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	@Qualifier("encounterService")
+	private EncounterService es;
+	
+	@Autowired
+	private EncounterRolesLoader loader;
+	
+	@Before
+	public void setup() {
+		
+		// an encounter type to retire
```
Comment (mseaton) [inline api/src/test/java/org/openmrs/module/initializer/api/EncounterRolesLoaderIntegrationTest.java:34]: Invalid comment

File: api/src/test/java/org/openmrs/module/initializer/api/EncounterRolesLoaderIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+		{
+			EncounterRole er = new EncounterRole();
+			er.setUuid("6eb3cf82-cae3-4d50-a31c-90f9230504c8");
+			er.setName("Plague Doctor");
+			es.saveEncounterRole(er);
+		}
+		// an encounter type to edit
```
Comment (mseaton) [inline api/src/test/java/org/openmrs/module/initializer/api/EncounterRolesLoaderIntegrationTest.java:41]: Invalid comment

**Review by mseaton — APPROVED** [review]
Generally looks good to me, just some minor edits needed from having used the encounterType domain as template you copied from...

**Comment by brandones** [comment]
Thanks @mseaton ! Fixed those.

File: api/src/test/java/org/openmrs/module/initializer/api/EncounterRolesLoaderIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+		}
+		// an encounter role to edit
+		{
+			EncounterRole er = new EncounterRole();
+			er.setUuid("09a55bcc-2ce2-46a2-836c-4a56dce9d239");
+			er.setName("Phlebotomist");
+			er.setDescription("Treats patients' phlegm");
+			es.saveEncounterRole(er);
+		}
+	}
```
Comment (mks-d) [inline api/src/test/java/org/openmrs/module/initializer/api/EncounterRolesLoaderIntegrationTest.java:49]: Would it be ok to add them through **test-metadata.xml**?

**Review by mks-d — COMMENTED** [review]
Thanks @brandones, looking perfect. Just a small comment.

**Review by mks-d — APPROVED** [review]

**Pull request merged**